tests/test_app/polls/middleware.py

```diff
@@ -0,0 +1,28 @@
+import random
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.models import User, Group
+from django.core.management import call_command
+from django.db import transaction
+from django.utils.deprecation import MiddlewareMixin
+
+
+class CleanupMiddleware(MiddlewareMixin):
+
+    @transaction.atomic
+    def process_response(self, request, response):
+        """
+        10% of requests will cause all data to be reset
+        """
+
+        if settings.PERIODIC_RESET:
+            if random.randint(1, 10) == 1:
```

Comment:
I think this is a bit too often... and also a bit too random 😄 

I would rather do this on a nightly cronjob, so people can go crazy with testing if they want to, without having to worry about all data magically resetting... What do you think?

Reply:
Noted, Ive removed this, and moved it to a nightly reset as recommended, we can adjust the frequency as necessary